### PR TITLE
Server technical debt

### LINKED ISSUE #519: Better for you to follow the stile guide: Server
## Description:
For the long time we didn't following the style, namely chapter _Names_ rule 2. It's time to change that!
This should remove such warnings:
```log
/home/runner/work/Juniorgram/Juniorgram/Network/Public/Include/Network/Primitives.hpp: In constructor ‘Network::ChannelInfo::ChannelInfo(uint64_t, uint64_t, const string&)’:
/home/runner/work/Juniorgram/Juniorgram/Network/Public/Include/Network/Primitives.hpp:32:9: warning: declaration of ‘channelName’ shadows a member of ‘Network::ChannelInfo’ [-Wshadow]
   32 |         : creatorID(creatorID), channelID(channelID), channelName(channelName)
      |     
```

## Acceptance Criteria:
Follow the rule and rename variables in proper way with underscore.

### PR DESCRIPTION
**Pull request by memorisecodead**
## Server Style: 
1. Fixing shadowVariables.
2. Adding new checking functions in Server class for Message Handler.
3. Updating constructor of Server.
4. Adding new function for accepting connect of Client.
5. Edited comments according to the style guide.
6. New comments added.

### CODE REVIEW

File: Server/Server.Core/Server.Core.Static/ArgumentParser.cpp [hunk 1]
```diff
@@ -108,5 +107,5 @@ void ArgumentParser::tryPushToMap(const std::string& key, const std::string& val
     if (!isInteger(value))
         throw std::runtime_error("key " + key + " has invalid value");
 
-    arguments.emplace(key, stoi(value));
-}
+    _arguments.try_emplace(key, stoi(value));
+}
```
Comment (Orange-hanter) [inline Server/Server.Core/Server.Core.Static/ArgumentParser.cpp]: fix

File: Server/Server.Core/Server.Core.Static/Server.hpp [hunk 2]
```diff
@@ -86,22 +91,156 @@ class Server
...
+     */
+    void directMessageCreateRequest(const std::shared_ptr<Connection>& client, Message& message) const;
+
+    /**
+     * @brief The method for getting default request.
+     */
+    void defaultRequest() const;
 };
-}  // namespace Server
+}  /// namespace Server
```
Comment (Orange-hanter) [inline Server/Server.Core/Server.Core.Static/Server.hpp]: fix

File: Server/Server.Core/Server.Core.Static/Server.cpp [hunk 3]
```diff
@@ -30,344 +31,126 @@ void Server::onClientDisconnect(const std::shared_ptr<Connection>& client)
...
-            msg.mHeader.mMessageType = Network::Message::MessageType::ServerMessage;
-            // msg << client->getID();
-
-            messageAllClients(msg, client);
+            readAllMessage(client, message);
         }
         break;
 
         case Network::Message::MessageType::ChannelListRequest:
         {
```
Comment (Orange-hanter) [inline Server/Server.Core/Server.Core.Static/Server.cpp:51]: Why do you need this bracket?

File: Server/Server.Core/Server.Core.Static/Server.cpp [hunk 4]
```diff
@@ -30,344 +31,126 @@ void Server::onClientDisconnect(const std::shared_ptr<Connection>& client)
...
         }
         break;
     }
 }
-
-Server::Server(const uint16_t& port)
-    : mAcceptor(mContext, asio::ip::tcp::endpoint(asio::ip::tcp::v4(), port)),
-      mPostgreManager(
-          std::make_unique<DataAccess::PostgreRepositoryManager>(DataAccess::PostgreAdapter::getInstance<DataAccess::PostgreAdapter>()))
+Server::Server(const uint16_t& port):
```
Comment (Orange-hanter) [inline Server/Server.Core/Server.Core.Static/Server.cpp]: Add space here

File: Server/Server.Core/Server.Core.Static/Server.cpp [hunk 5]
```diff
@@ -382,80 +165,92 @@ void Server::start()
...
     for (size_t i = 0; i < threadsCount; ++i)
     {
-        mThreads.emplace_back(std::thread([this]() { mContext.run(); }));
+        _threads.emplace_back(std::thread([this]() { _context.run(); }));
     }
-    mPostgreManager->handleRequests();
+    _postgreManager->handleRequests();
 
-    Base::Logger::FileLogger::getInstance().log("[SERVER] Started!", Base::Logger::LogLevel::INFO);
+    Base::Logger::FileLogger::getInstance().log
```
Comment (Orange-hanter) [inline Server/Server.Core/Server.Core.Static/Server.cpp]: Also, it's a good idea to use some alias for logger instead so long line

File: Server/Server.Core/Server.Core.Static/Server.cpp [hunk 6]
```diff
@@ -382,80 +165,92 @@ void Server::start()
...
+    (
+        "[SERVER] Stopped!",
+        Base::Logger::LogLevel::INFO
+    );
 }
 
 void Server::waitForClientConnection()
+{   
+    _acceptor.async_accept([this](const std::error_code error, asio::ip::tcp::socket socket) { acceptingClientConnection(error,socket); });
```
Comment (Orange-hanter) [inline Server/Server.Core/Server.Core.Static/Server.cpp]: Why you didn't create a separate functor variable and not set it like argument value?

File: Server/Server.Core/Server.Core.Static/Server.cpp [hunk 7]
```diff
@@ -382,80 +165,92 @@ void Server::start()
...
-                );
-            }
         }
         else
         {
             Base::Logger::FileLogger::getInstance().log
             (
-                "[SERVER] New Connection Error: " + error.message(),
-                Base::Logger::LogLevel::ERR
+                "[-----] Connection Denied",
```
Comment (Orange-hanter) [inline Server/Server.Core/Server.Core.Static/Server.cpp]: Why dashes?

File: Server/Server.Core/Server.Core.Static/Server.cpp [hunk 8]
```diff
@@ -500,32 +295,379 @@ void Server::messageAllClients(const Message& message, const std::shared_ptr<Con
...
+    Message messageHandler;
+    messageHandler.mHeader.mMessageType = Message::MessageType::ServerMessage;
+    
+    // msg << client->getID();
+    messageAllClients(messageHandler, client);
+}
+
+void Server::channelListRequest(const std::shared_ptr<Connection>& client) const
+{
+    using namespace DataAccess;
```
Comment (Orange-hanter) [inline Server/Server.Core/Server.Core.Static/Server.cpp]: Move this in all CPP scope. Do not duplicate code

File: Server/Server.Core/Server.Core.Static/Server.cpp [hunk 9]
```diff
@@ -500,32 +295,379 @@ void Server::messageAllClients(const Message& message, const std::shared_ptr<Con
...
+}
+void Server::defaultRequest() const
+{ 
+    Base::Logger::FileLogger::getInstance().log
+    (
+        "Unknown command received", 
+        Base::Logger::LogLevel::ERR
+    );
+}
+}  /// namespace Server
```
Comment (Orange-hanter) [inline Server/Server.Core/Server.Core.Static/Server.cpp]: fix

File: Server/Server.Core/Server.Core.Static/Server.cpp [hunk 10]
```diff
@@ -500,32 +295,379 @@ void Server::messageAllClients(const Message& message, const std::shared_ptr<Con
...
+
+    messageToClient.mBody = std::make_any<Utility::DirectMessageStatus>(futureResult.get());
+    client->send(messageToClient);
+}
+void Server::defaultRequest() const
+{ 
+    Base::Logger::FileLogger::getInstance().log
+    (
+        "Unknown command received", 
+        Base::Logger::LogLevel::ERR
```
Comment (Orange-hanter) [inline Server/Server.Core/Server.Core.Static/Server.cpp]: Make shorter, applying LogLevel namespace to entire CPP

File: Server/Server.Core/Server.Core.Static/Server.cpp (hunk 3 above)
Comment (memorisecodead) [reply Server/Server.Core/Server.Core.Static/Server.cpp:51]: image

File: Server/Server.Core/Server.Core.Static/Server.cpp (hunk 7 above)
Comment (memorisecodead) [reply Server/Server.Core/Server.Core.Static/Server.cpp]: image

I don't know, i don't touched it. 
It was in dev branch.

Comment (memorisecodead) [reply Server/Server.Core/Server.Core.Static/Server.cpp]: > 

I'm resolve it in my branch some of result
image

File: Server/Server.Core/Server.Core.Static/Server.cpp (hunk 3 above)
Comment (Orange-hanter) [reply Server/Server.Core/Server.Core.Static/Server.cpp:51]: ???? 
What should I see on your picture??

Comment (memorisecodead) [reply Server/Server.Core/Server.Core.Static/Server.cpp:51]: image

Comment (Orange-hanter) [reply Server/Server.Core/Server.Core.Static/Server.cpp:51]: i`m asking you about case brackets
case ....
{  <- this one
....
} <- and this one

Comment (memorisecodead) [reply Server/Server.Core/Server.Core.Static/Server.cpp:51]: image
image
I did this as we have it in SerializationHandler.hpp. It's purely a visual style.

File: Server/Server.Core/Server.Core.Static/Server.cpp [hunk 11]
```diff
@@ -664,10 +630,10 @@ void Server::directMessageCreateRequest(const std::shared_ptr<Connection>& clien
...
+    FileLogger::getInstance().log
     (
         "Unknown command received", 
-        Base::Logger::LogLevel::ERR
+        LogLevel::ERR
     );
 }
-}  /// namespace Server
\ No newline at end of file
+}  // namespace Server
```
Comment (Orange-hanter) [inline Server/Server.Core/Server.Core.Static/Server.cpp]: Fix end line, please

Comment (Orange-hanter) [reply Server/Server.Core/Server.Core.Static/Server.cpp]: In all files

File: Server/Server.Core/Server.Core.Static/Server.cpp (hunk 3 above)
Comment (Orange-hanter) [reply Server/Server.Core/Server.Core.Static/Server.cpp:51]: Excuse me, but I didn't see any brackets at SerializationHandler.hpp

File: Server/Server.Core/Server.Core.Static/Server.cpp (hunk 11 above)
Comment (memorisecodead) [reply Server/Server.Core/Server.Core.Static/Server.cpp]: I did it. In our files we have these example:
image
My decision as the same as we have:
image
Or what you mean?


> In all files

File: Server/Server.Core/Server.Core.Static/Server.cpp (hunk 3 above)
Comment (memorisecodead) [reply Server/Server.Core/Server.Core.Static/Server.cpp:51]: It's the same, but keyword "break" after brackets :/

Comment (memorisecodead) [reply Server/Server.Core/Server.Core.Static/Server.cpp:51]: For example, also in dev branch we have this brackets: https://github.com/f1nal3/Juniorgram/blob/dev/Server/Server.Core/Server.Core.Static/Server.cpp


> Excuse me, but I didn't see any brackets at SerializationHandler.hpp

File: Server/Server.Core/Server.Core.Static/Server.cpp (hunk 11 above)
Comment (Orange-hanter) [reply Server/Server.Core/Server.Core.Static/Server.cpp]: Hmmm. What do I mean by that....
image

File: Server/Server.Core/Server.Core.Static/Server.cpp (hunk 3 above)
Comment (Orange-hanter) [reply Server/Server.Core/Server.Core.Static/Server.cpp:51]: And you don't see any difference between these two cases?

Comment (memorisecodead) [reply Server/Server.Core/Server.Core.Static/Server.cpp:51]: > And you don't see any difference between these two cases?

The second one is made using Network, and it also takes out the states from each function.

**Comment by Orange-hanter** [comment]
@memorisecodead what happend?

**Comment by memorisecodead** [comment]
> @memorisecodead what happend?

Because the port is initialized in the main with arguments. If we null it, we won't connect to the server.

**Pull request closed without merging**